jsbits/Scenario.js

```diff
@@ -192,10 +194,22 @@ function gm$smartProcessFeatureCollection(fc, defVec, maxGeomId) {
     points.forEach(addGeomID);
     lines.forEach(addGeomID);
     surfaces.forEach(addGeomID);
+
+    // Get lat, lon, alt, srid if available
+    if (gi['lat'] && gi['lat'].constructor === Number &&
+        gi['lon'] && gi['lon'].constructor === Number &&
+        gi['alt'] && gi['alt'].constructor === Number) {
+        lat = gi['lat'];
+        lon = gi['lon'];
+        alt = gi['alt'];
+    }
+    if (gi['srid'] && gi['srid'].constructor === Number) {
+        srid = gi['srid']
+    }
+
     // transform everything from WGS84 to a metric reference system if needed
-    if(cmin.length >= 2 && cmax.length >= 2
-       && cmin[0] > -360 && cmax[0] < 360 && cmin[1] > -180 && cmax[1] < 180
-       && (cmax[0] - cmin[0]) < 5 && (cmax[1] - cmin[1]) < 5) {
```

Comment:
The case where the heuristic fail is when we previously upload a model of one bicycle in metric system with length shorter than 5m, and the heuristic pass it as WGS'84. I think we can improve the heuristic in line 198 so that, if the number of object is less than 10 (usually Johannes like to test model he just made), then the threshold is 1m, if the number of object is less than 100, the threshold is 3m, else the threshold is 5m. Something similar to this.